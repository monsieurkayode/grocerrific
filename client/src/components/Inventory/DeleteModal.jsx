import React from 'react';
import { func, string, bool } from 'prop-types';

import Modal from '../common/Modal';

const DeleteModal = ({
  id,
  name,
  closeModal,
  handleDelete,
  deleting
}) => (
  <Modal
    title={`Delete ${name}`}
    closeModal={closeModal}
  >
    <div className="notice">
      <p>Are you sure? This action cannot be undone!</p>
    </div>
    <div className="checkout__button">
      <button
        onClick={() => handleDelete(id)}
        className="buy uppercase"
        type="button"
        disabled={deleting}
      >
        {deleting ? 'Deleting...' : 'Confirm'}
      </button>
    </div>
  </Modal>
);

DeleteModal.propTypes = {
  id: string.isRequired,
  name: string.isRequired,
  deleting: bool.isRequired,
  closeModal: func.isRequired,
  handleDelete: func.isRequired
};

export default DeleteModal;
