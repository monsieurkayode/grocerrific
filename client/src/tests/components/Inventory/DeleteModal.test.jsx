import React from 'react';
import { shallow } from 'enzyme';
import DeleteModal from '../../../components/Inventory/DeleteModal';

const setup = () => {
  const props = {
    id: '1',
    name: 'Grocery',
    closeModal: jest.fn(),
    handleDelete: jest.fn(),
    deleting: false
  };
  const wrapper = shallow(<DeleteModal {...props} />);

  return { wrapper, props };
};

describe('DeleteModal Component', () => {
  const { wrapper, props } = setup();

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleDelete when the delete button is clicked', () => {
    const deleteButton = wrapper.find('button');
    deleteButton.simulate('click');
    expect(props.handleDelete).toHaveBeenCalled();
    expect(props.handleDelete).toHaveBeenCalledWith('1');
  });

  it('should set the text on the delete button to Deleting... when deleting',
    () => {
      wrapper.setProps({ deleting: true });
      const deleteButton = wrapper.find('button');
      expect(deleteButton.text()).toBe('Deleting...');
    });
});
