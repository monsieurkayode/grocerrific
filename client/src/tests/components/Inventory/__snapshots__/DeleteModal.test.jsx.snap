// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteModal Component renders correctly 1`] = `
<Modal
  closeModal={[MockFunction]}
  title="Delete Grocery"
>
  <div
    className="notice"
  >
    <p>
      Are you sure? This action cannot be undone!
    </p>
  </div>
  <div
    className="checkout__button"
  >
    <button
      className="buy uppercase"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Confirm
    </button>
  </div>
</Modal>
`;
